Framework/Crystal/src/AnvredCorrection.cpp

```diff
@@ -168,30 +166,21 @@ void AnvredCorrection::exec() {
   const V3D pos = m_inputWS->getInstrument()->getSource()->getPos() - samplePos;
   double L1 = pos.norm();
 
+  const auto &spectrumInfo = m_inputWS->spectrumInfo();
+
   Progress prog(this, 0.0, 1.0, numHists);
   // Loop over the spectra
   PARALLEL_FOR_IF(Kernel::threadSafe(*m_inputWS, *correctionFactors))
   for (int64_t i = 0; i < int64_t(numHists); ++i) {
     PARALLEL_START_INTERUPT_REGION
 
-    // Get detector position
-    IDetector_const_sptr det;
-    try {
-      det = m_inputWS->getDetector(i);
-    } catch (Exception::NotFoundError &) {
-      // Catch if no detector. Next line tests whether this happened - test
-      // placed
-      // outside here because Mac Intel compiler doesn't like 'continue' in a
-      // catch
-      // in an openmp block.
-    }
-    // If no detector found, skip onto the next spectrum
-    if (!det)
+    // If no detector is found, skip onto the next spectrum
+    if (!spectrumInfo.hasDetectors(i))
       continue;
 
     // This is the scattered beam direction
     Instrument_const_sptr inst = m_inputWS->getInstrument();
```

Comment:
I though the variable was unused now, but actually it is still used, so ignore my comment.